packages/go_router/lib/src/route.dart

```diff
@@ -55,169 +45,97 @@ class GoRoute {
-  ///
-  /// If there are multiple routes that match the location, the first match is used.
-  /// To make predefined routes to take precedence over dynamic routes eg. '/:id'
-  /// consider adding the dynamic route at the end of the routes
-  /// For example:
-  /// ```
-  /// final GoRouter _router = GoRouter(
-  ///   routes: <GoRoute>[
-  ///     GoRoute(
-  ///       path: '/',
-  ///       redirect: (_) => '/family/${Families.data[0].id}',
-  ///     ),
-  ///     GoRoute(
-  ///       path: '/family',
-  ///       pageBuilder: (BuildContext context, GoRouterState state) => ...,
-  ///     ),
-  ///     GoRoute(
-  ///       path: '/:username',
-  ///       pageBuilder: (BuildContext context, GoRouterState state) => ...,
-  ///     ),
-  ///   ],
-  /// );
-  /// ```
-  /// In the above example, if /family route is matched, it will be used.
-  /// else /:username route will be used.
-  final List<GoRoute> routes;
-
-  /// An optional redirect function for this route.
-  ///
-  /// In the case that you like to make a redirection decision for a specific
-  /// route (or sub-route), you can do so by passing a redirect function to
-  /// the GoRoute constructor.
-  ///
-  /// For example:
-  /// ```
-  /// final GoRouter _router = GoRouter(
-  ///   routes: <GoRoute>[
-  ///     GoRoute(
-  ///       path: '/',
-  ///       redirect: (_) => '/family/${Families.data[0].id}',
-  ///     ),
-  ///     GoRoute(
-  ///       path: '/family/:fid',
-  ///       pageBuilder: (BuildContext context, GoRouterState state) => ...,
-  ///     ),
-  ///   ],
-  /// );
-  /// ```
-  final GoRouterRedirect redirect;
+  /// Extract the path parameters from a match.
+  Map<String, String> extractPathParams(RegExpMatch match) =>
+      extractPathParameters(_pathParams, match);
 
   /// Match this route against a location.
   RegExpMatch? matchPatternAsPrefix(String loc) =>
       _pathRE.matchAsPrefix(loc) as RegExpMatch?;
 
-  /// Extract the path parameters from a match.
-  Map<String, String> extractPathParams(RegExpMatch match) =>
-      extractPathParameters(_pathParams, match);
+  static String? _emptyRedirect(GoRouterState state) => null;
+}
+
+/// Route configuration object equivalent to [StackedRoute].
+class GoRoute extends StackedRoute {
+  /// Constructs a [GoRoute] object.
+  GoRoute({
+    required super.path,
+    super.builder,
+    super.pageBuilder,
+    super.redirect,
+    super.routes,
+    super.name,
+  });
+}
+
+/// A route that is displayed visually above the matching parent route using the
+/// [Navigator].
+///
+/// The widget returned by [builder] is wrapped in [Page] and provided to the
+/// root Navigator or the Navigator belonging to the nearest [NestedStackRoute]
+/// ancestor. The page will be either a [MaterialPage] or [CupertinoPage]
+/// depending on the application type.
+///
+/// This route has the same behavior as GoRoute in go_router >=3.0.0.
+class StackedRoute extends RouteBase {
+  /// Constructs a [StackedRoute].
```

Comment:
can you add some information about the parameter assertion?

Reply:
Sure